apps/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,45 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
+import { Todo } from './models/interface.todo';
+import { TodosService } from './services/todos.service';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { ComponentStore } from '@ngrx/component-store';
+import { TodosStore } from './store/todos.store';
+import { ItemComponent } from './components/item.component';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule, ItemComponent],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
+    <div *ngIf="todos$ | async; else loading">
+      <div *ngFor="let todo of todos$ | async">
+        <app-item
+          [todo]="todo"
+          [processingItemId]="processingItemId$ | async"></app-item>
+      </div>
     </div>
+    <ng-template #loading>
+      <mat-spinner></mat-spinner>
+    </ng-template>
   `,
   styles: [],
+  providers: [ComponentStore, TodosStore],
 })
 export class AppComponent implements OnInit {
-  todos!: any[];
+  todos$ = this.todosStore.todos$;
+  processingItemId$ = this.todosStore.processingItemId$; // Get processingItemId from TodosStore
 
-  constructor(private http: HttpClient) {}
+  constructor(
+    public todosService: TodosService,
+    private readonly todosStore: TodosStore
+  ) {}
 
   ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
-  }
-
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
-        }
-      )
-      .subscribe((todoUpdated: any) => {
-        this.todos[todoUpdated.id - 1] = todoUpdated;
-      });
+    this.todosService.initTodos().subscribe({
+      next: (todos: Todo[] | null) => {
+        this.todosStore.loadTodos(todos);
+      },
+    });
```

Comment:
Be careful with manual subscribe

Ngrx updater accept observable as input and they handle the subscription for you.